app/main.py

```diff
@@ -1,3 +1,73 @@
+from __future__ import annotations
+from typing import Optional
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance" | int | float) -> Optional["Distance"]:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
```

Comment:
```suggestion
        if isinstance(other, Distance):
            return Distance(self.km + other.km)
        return Distance(self.km + other)
```
Maybe here and in next situations we can write code withoute second if, because we already have anotations of parameters

Reply:
Good advice, thanks )